be2-scala/src/main/scala/ch/epfl/pop/pubsub/graph/validators/SocialMediaValidator.scala

```diff
@@ -22,10 +25,17 @@ case object SocialMediaValidator extends MessageDataContentValidator with EventV
             case Some(message) => {
                 val data: AddChirp = message.decodedData.get.asInstanceOf[AddChirp]
 
-                val actualSender: PublicKey = message.sender //sender's PK
+                val sender: PublicKey = message.sender //sender's PK
+                val channel: Channel = rpcMessage.getParamsChannel
 
                 if (!validateTimestampStaleness(data.timestamp)) {
                     Right(validationError(s"stale timestamp (${data.timestamp})"))
+                } else if (!validateChannelType(ObjectType.CHIRP, channel)){
+                    Right(validationError(s"trying to add a Chirp on a wrong type of channel $channel"))
+                } else if (!validateAttendee(sender, channel)){
+                    Right(validationError(s"invalid PoP token $sender"))
```

Comment:
A more expressive message and provide a reason of why the PoP token is invalid. Same for below